src/components/ApplicationGroup/Details/EnvironmentOverview/EnvironmentOverview.tsx

```diff
@@ -82,61 +166,292 @@ export default function EnvironmentOverview({
     }
 
     const renderAppInfoRow = (item: AppInfoListType, index: number) => {
+        const isSelected = selectedAppIds.includes(item.appId)
         return (
             <div
                 key={`${item.application}-${index}`}
-                className="app-deployments-info-row display-grid dc__align-items-center"
+                className={`app-deployments-info-row dc__w-fit-inherit display-grid dc__align-items-center ${
+                    isHovered === index ? 'bc-n50' : 'bcn-0'
+                } ${lastDeployedClassName}`}
+                onMouseEnter={() => setIsHovered(index)}
+                onMouseLeave={() => setIsHovered(null)}
             >
-                <span className="fs-13 fw-4 cn-7">{item.application}</span>
-                {!isVirtualEnv && (
-                    <AppStatus
-                        appStatus={item.lastDeployed ? item.appStatus : StatusConstants.NOT_DEPLOYED.noSpaceLower}
-                    />
-                )}
+                <div
+                    className={`pl-16 pr-16 app-deployment-info-row-leftsection dc__align-items-center display-grid dc__position-sticky sticky-column ${
+                        isHovered === index ? 'bc-n50' : 'bcn-0'
+                    }`}
+                >
+                    {/* <span> */}
+                    {isHovered !== index && !isSelected ? (
+                        <DevtronIcon className="icon-dim-20" />
+                    ) : (
+                        <label className="dc__position-rel pointer mb-0">
+                            <input
+                                type="checkbox"
+                                className="form__checkbox"
+                                value={item.appId}
+                                onChange={handleSelect}
+                                checked={isSelected}
+                            />
+                            <span className={`form__checkbox-container ${isSelected ? 'tick-icon' : ''}`}></span>
+                        </label>
+                    )}
+                    {/* </span> */}
+                    {!isVirtualEnv && <AppStatus appStatus={item.appStatus} hideStatusMessage />}
+                    <span className="fs-13 fw-4 cn-7 dc__ellipsis-right">{item.application}</span>
+                </div>
                 <AppStatus
```

Comment:
The AppStatus in the dashboard is being deleted. Pls use it from common lib. Same for processDeployedTime

Reply:
is it deleted already in main ? @eshankvaish

Reply:
The change is there on the common lib. It will be deleted in main with one of my PR. You would just need to update the import

Reply:
will that pr be merged today?